common/vcs/git.go

```diff
@@ -0,0 +1,204 @@
+/*
+ *  Copyright IBM Corporation 2021
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package vcs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/sirupsen/logrus"
+)
+
+// GitVCSRepo stores git repo config
+type GitVCSRepo struct {
+	InputURL       string
+	URL            string
+	Branch         string
+	Tag            string
+	CommitHash     string
+	PathWithinRepo string
+	GitRepository  *git.Repository
+	GitRepoPath    string
+}
+
+func isGitCommitHash(commithash string) bool {
+	gitCommitHashRegex := regexp.MustCompile(`^[a-fA-F0-9]{40}$`)
+	return gitCommitHashRegex.MatchString(commithash)
+}
+
+func isGitBranch(branch string) bool {
+	gitBranchRegex := regexp.MustCompile(`^(main|development|master|(features|tests|(bug|hot)fix)((\/|-)[a-zA-Z0-9]+([-_][a-zA-Z0-9]+)*){1,2}|release(\/|-)[0-9]+(\.[0-9]+)*(-(alpha|beta|rc)[0-9]*)?)$`)
+	return gitBranchRegex.MatchString(branch)
+}
+
+func isGitTag(tag string) bool {
+	gitTagRegex := regexp.MustCompile(`^v[0-9]+(\.[0-9]+)?(\.[0-9]+)?$`)
+	return gitTagRegex.MatchString(tag)
+}
```

Comment:
Used regex for effective matching and extraction of parts of information from the input.